File: gdscript/src/main/java/gdscript/psi/impl/GdInheritanceSubIdRefImpl.java
package gdscript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import gdscript.psi.GdInheritanceSubIdRef;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

public class GdInheritanceSubIdRefImpl extends GdRefElementImpl implements GdInheritanceSubIdRef {

  public GdInheritanceSubIdRefImpl(ASTNode node) {
    super(node);
  }

  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitInheritanceSubIdRef(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

}

File: gdscript/src/main/kotlin/gdscript/psi/impl/GdInheritanceSubIdRefImpl.kt
package gdscript.psi.impl

import com.intellij.lang.ASTNode
import com.intellij.model.psi.PsiSymbolReference
import com.intellij.polySymbols.references.polySymbolOwnReferences
import com.intellij.psi.PsiElementVisitor
import gdscript.polySymbols.GdPolySymbolKind.INHERITANCE_SYMBOLS
import gdscript.psi.GdInheritanceSubIdRef
import gdscript.psi.GdVisitor
import org.jetbrains.annotations.Unmodifiable

class GdInheritanceSubIdRefImpl(node: ASTNode) : GdRefElementImpl(node), GdInheritanceSubIdRef {
    fun accept(visitor: GdVisitor) {
        visitor.visitInheritanceSubIdRef(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override fun getOwnReferences(): @Unmodifiable Collection<PsiSymbolReference> =
        polySymbolOwnReferences(this) {
            resolveFromNameMatchQuery(INHERITANCE_SYMBOLS, text)
        }
}
